### DISCUSSION TITLE
몇가지 질문이 있습니다!!

### DISCUSSION THREAD

**Discussion by tjrkdgnl** [solvedac/unofficial-documentation#34] [질답] [answered]
API 문서를 보다가 원하는 기능이 없는 것 같아서 질문드립니다!

1. 이용자가 못푼 문제 IDs를 얻어오는 API는 없을까요? search 의 문제 검색하기 API에서 query를 이용하면 이용자가 풀어본 목록을 가져올 수 있던데 이처럼 이용자가 못푼 문제도 가져올 수 있도록 query를 작성할 수 있나요?

2. 이용자가 질문을 남긴 문제, 질문에 대한 답글을 가져올 수 있는 API는 없을까요?

3. IDs를 통해 문제를 검색하면 response로 문제 지문, test case 등에 대한 값은 주지 않나요?

**Comment by RanolP** [comment, marked as answer]
1. not 연산과 solved_by 필터를 이용해서 검색하시면 됩니다.
  고급 검색 가이드는 [여기](https://solved.ac/search)에서 찾아보실 수 있습니다.
    예시로는 `-solved_by:shiftpsh` 를 들 수 있겠습니다.
2. solved.ac는 문제 난이도 제공 서비스이고, 해당 기능을 구현하고 있지 않습니다.
  해당 기능을 원하신다면 StartLink 측에 직접 문의하시는 편이 가장 좋고, 차선책으로는 웹 스크래핑 등의 방법이 있습니다.
  다만, 웹 스크래핑의 경우 공식적으로 이용 규칙에서 금지한 행위이므로 권장하지는 않습니다.
3. 위와 같음
